### DISCUSSION TITLE
my question is why in lua?

### DISCUSSION THREAD

**Discussion by njavilas** [OptimalBits/bull#2227] [Q&A]
Good evening, I am delighted with the library and I am studying it to take advantage of all the tools it offers. But I have a question because the commands are in lua? it was not better to do it in js I ask for ignorance not for criticism. has higher performance or is a migration from another project

**Comment by manast** [comment]
Because lua is the language currently supported by Redis for writing atomic scripts, which is the only way to provide the functionality that Bull provides as robust as it does. Btw, if you are new to Bull you probably want to start using BullMQ instead: 
https://github.com/taskforcesh/bullmq and documentation here: https://docs.bullmq.io/
